P202.Training.Test/RoleServiceTest.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Moq;
+using P202.Training.Data.Entities;
+using P202.Training.Data.Repositories;
+using P202.Training.Domain;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace P202.Training.Test
+{
+    public class RoleServiceTest
+    {
+        [Fact]
+        public void RoleService_CreateData()
+        {
+            var newRole = new Role();
+            var mockpersonRepository = new Mock<IRoleRepository>();
+            mockpersonRepository.Setup(x => x.CreateRole(newRole)); 
+            mockpersonRepository.Object.CreateRole(newRole);
```

Comment:
For the class name it seems your intention is to test the `RoleService` but you're never calling any of it's methods. You should instantiate a `RoleService` passing the mock as a parameter and then test it's methods. Something like this:
```
var roleService = new RoleService(mockRoleRepository.Object);
roleService.CreateRole(...);

// Assertions
```

Reply:
I'm not sure we should have UT for the repository at all. If we wanted to test it, maybe they should be integration tests, and thus we shouldn't mock the repository methods.

Reply:
@martinkeimel  I'll apply the solution that you suggest.